### PR TITLE
tls: remove obsolete openssl version check and fix libressl build

### PR DESCRIPTION
**Pull request by cspiel1**
Solves: https://github.com/baresip/re/issues/678

### CODE REVIEW

**Comment by cspiel1** [comment]
Two more version checks have to be removed.

**Comment by cspiel1** [comment]
Removed the `OPENSSL_VERSION_NUMBER >= 0x10100000L` check completely.

File: src/md5/wrap.c [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (sreimers) [inline src/md5/wrap.c:1]: unrelated change

File: src/tls/openssl/tls.c [hunk 2]
```diff
@@ -1570,10 +1546,9 @@ static int tls_session_update_cache(const struct tls_conn *tc,
 					     session_cmp_handler, &peer));
 	mem_deref(e);
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10101000L) && \
```
Comment (sreimers) [inline src/tls/openssl/tls.c:1573]: This check is needed 0x10101000L is openssl 1.1.1

File: src/tls/openssl/tls.c [hunk 3]
```diff
@@ -1596,7 +1571,6 @@ static int tls_session_update_cache(const struct tls_conn *tc,
 }
 
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
```
Comment (sreimers) [inline src/tls/openssl/tls.c:1599]: This check is needed 0x10101000L is openssl 1.1.1

File: src/tls/openssl/tls.c [hunk 4]
```diff
@@ -1680,14 +1653,9 @@ int tls_set_session_reuse(struct tls *tls, int enabled)
 	if (!tls->reuse.enabled)
 		return 0;
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
```
Comment (sreimers) [inline src/tls/openssl/tls.c:1683]: This check is needed 0x10101000L is openssl 1.1.1

File: src/tls/openssl/tls.c [hunk 5]
```diff
@@ -1703,11 +1671,7 @@ bool tls_session_reused(const struct tls_conn *tc)
 	if (!tc)
 		return false;
 
-#if (OPENSSL_VERSION_NUMBER >= 0x10101000L)
```
Comment (sreimers) [inline src/tls/openssl/tls.c:1706]: This check is needed 0x10101000L is openssl 1.1.1

File: src/tls/openssl/tls.c (hunk 2 above)
Comment (cspiel1) [reply src/tls/openssl/tls.c:1573]: I guess also in tls.c:61
```diff
--- b/src/tls/openssl/tls.c
+++ a/src/tls/openssl/tls.c
@@ -61,7 +61,7 @@ struct tls_cert {
 	char *host;
 };
 
-#if defined(TRACE_SSL)
+#if defined(TRACE_SSL) && (OPENSSL_VERSION_NUMBER >= 0x10101000L)
 /**
  * Global flag if key material must be appended to file
  */
```

File: src/tls/openssl/tls.c [hunk 6]
```diff
@@ -1905,6 +1884,12 @@ int tls_add_certf(struct tls *tls, const char *certf, const char *host)
 	}
 
 	return err;
+#else
+	(void)tls;
+	(void)certf;
+	(void)host;
+	return 0;
```
Comment (sreimers) [inline src/tls/openssl/tls.c:1891]: Maybe better to return ENOSYS here.
